### DISCUSSION TITLE
Throw when no source property found for destination

### DISCUSSION THREAD

**Discussion by EniacMlezi** [riok/mapperly#295] [Ideas]
It would be great to have a way to force all destination members to be mapped or explicitly ignored, like [mapster does](https://github.com/MapsterMapper/Mapster/wiki/Config-validation-&-compilation#checking-destination-member)

```cs
public class Person
{
    public required int Id { get; init; }

    public required string Name { get; set; }
}

public class PersonDto
{
    public required int Id { get; init; }

    public required string Name { get; set; }

    public int something { get; set; }
}

[Mapper(**ThrowOnNoSourcePropertyFound = true**)]
public partial class PersonMapper
{
    public partial PersonDto PersonToPersonDto(Person person);

}
```

When mapping from `Person` to `PersonDto`, no source property for  '`something`' is found, so the mapper throws.

**Comment by latonz** [comment]
This should already be possible by setting the severity of the diagnostics `RMG012` and `RMG020 ` to error (see [strict mappings](https://mapperly.riok.app/docs/configuration/analyzer-diagnostics#strict-mappings)).

**Reply by EniacMlezi** [reply]
Ow did not realize, awesome!
